core/src/org/javarosa/core/model/instance/TreeReference.java

```diff
@@ -337,18 +442,19 @@ else if (!this.getInstanceName().equals(contextRef.getInstanceName())) {
         TreeReference newRef = anchor(contextRef);
         newRef.setContext(contextRef.getContext());
 
-        //apply multiplicites and fill in wildcards as necessary based on the context ref
+        // apply multiplicites and fill in wildcards as necessary, based on the
+        // context ref
         for (int i = 0; i < contextRef.size() && i < newRef.size(); i++) {
-
-            //If the the contextRef can provide a definition for a wildcard, do so
-            if (TreeReference.NAME_WILDCARD.equals(newRef.getName(i)) && !TreeReference.NAME_WILDCARD.equals(contextRef.getName(i))) {
+            // If the the contextRef can provide a definition for a wildcard, do so
+            if (TreeReference.NAME_WILDCARD.equals(newRef.getName(i)) &&
+                    !TreeReference.NAME_WILDCARD.equals(contextRef.getName(i))) {
                 newRef.data.setElementAt(newRef.data.elementAt(i).setName(contextRef.getName(i)), i);
             }
 
             if (contextRef.getName(i).equals(newRef.getName(i))) {
-                //We can't actually merge nodes if the newRef has predicates or filters
-                //on this expression, since those reset any existing resolutions which
-                //may have been done.
+                // We can't actually merge nodes if the newRef has predicates
+                // or filters on this expression, since those reset any
+                // existing resolutions which may have been done.
                 if (newRef.getPredicate(i) == null) {
```

Comment:
Pretty sure it is just to remove the whole multiplicity copying logic. The call to anchor correctly copies/merges the multiplicity values, and this call to setMultiplicity messes them up if they have the same name but are in fact different parts of the reference.